### DISCUSSION TITLE
model-viewer ar only opens fullscreen for my glb file (not ar)

### DISCUSSION THREAD

**Discussion by Asthenean** [google/model-viewer#2195] [Q&A]
### Description

I am trying to use the AR viewer button with my model, but it only appears to open the full-screen Scene Viewer with a black background.
File: https://nickel-pushy-iris.glitch.me/assets/babylonexport.glb

When replaced by the astronaut from the demo here it does work:
https://model-viewer.glitch.me/

https://arvr.google.com/scene-viewer-preview
EDIT:  export with 3ds Max and Babylon.js and fixing filesize and uv we have no errors in the scene viewer anymore and it the file passes the validator. But the scene viewer still opens fullscreen with a black background...

#### Live Demo
https://nickel-pushy-iris.glitch.me/

### Browser Affected

- [x] Chrome
- [ ] Edge
- [x] Firefox
- [ ] Helios
- [ ] IE
- [ ] Safari

### OS

- [x] Android
- [ ] iOS
- [ ] Linux
- [ ] MacOS
- [ ] Windows

### Versions

- model-viewer: v0.9.0
- three.js: ?

  webcomponentsjs@2.1.3
  intersection-observer@0.5.1
  resize-observer-polyfill@1.5.0
  fullscreen-polyfill@1.0.2
  @magicleap/prismatic
  focus-visible@5.0.2

**Comment by elalish** [comment]
@efsjouw That's a SceneViewer error, so I'll forward it to @tpsiaki. Meanwhile, please check your glTF here, as that may give you more actionable feedback on what may be wrong with it, if anything: https://github.khronos.org/glTF-Validator/

**Comment by Asthenean** [comment]
Alright the validator gives me a similar message.

Also some UNUSED_OBJECT messages
Vector3 at accessor indices 506376..506378 is not of unit length: 0.6179051427527715.

I will look into this, but I am not sure what these errors mean right now.

**Comment by elalish** [comment]
@efsjouw What software exported the glTF? It would be good to file a bug on them as well with the validator messages. Are there errors or only warnings? It's a little surprising that our parser can handle it properly if the glTF is actually invalid.

**Comment by Asthenean** [comment]
It was exported in Blender using the standard settings.

EDIT - We redid the export with 3ds Max and Babylon.js and now it passes the validator but still shows up with a black background in the viewer.

**Comment by Asthenean** [comment]
I know this is tagged as 'not model-viewer' but where am I supposed to leave this issue, as I understand scene viewer is closed source? So this problem is not solvable?

The only thing it does is open the model with a black background without any errors, would logcat show me any info?

**Comment by jackmez** [comment]
I found a workaround by opening the model I had exported from the 3ds max babylon exporter here:
https://sandbox.babylonjs.com/

then re-exporting as a .glb file.

This removed the errors in the file and allowed me to open on android without the black background.

**Comment by Asthenean** [comment]
That doesn't appear to work for me, also it seems to make the file much larger. 
I told our modeller to get rid of all the errors, but I do see there are still some warnings left.
Did you get rid of all errors/warnings?
